frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -1517,14 +1526,23 @@ func dispatchCopy(d *dispatchState, cfg copyConfig) error {
 			if gitRef.SubDir != "" {
 				commit += ":" + gitRef.SubDir
 			}
-			gitOptions := []llb.GitOption{llb.WithCustomName(pgName)}
+			gu := gitRef.GitURL()
+			fullURL := src
+			if gu.Remote != "" {
+				fullURL = gu.Remote
+			}
+			gitOptions := []llb.GitOption{llb.WithCustomName(pgName), llb.GitChecksum(gitRef.Checksum),
```

Comment:
I don't quite get why do we need to call `ParseGitRef` and then still pass a bunch of the options manually. Why can't we convert to a structure that already can be connected to `llb`. There are so many duplicates in here: user needs to pass parsed `GitURLBase` structure, but then still needs to pass `GitFullURL` as well. Why is the commit passed to both `GitChecksum` and `GitIDSuffix`. These look like internal behaviors what the user should not know about. If we parse the input URL we should have remote(full), ref, commit, subdir and we should be able to pass them without duplicates.